egs2/TEMPLATE/ssl1/ssl.sh

```diff
@@ -0,0 +1,585 @@
+
+# Copyright 2021 Tianzi Wang
+#           2022 Xuankai Chang
+#           2025 William Chen
+# Apache 2.0
+
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+min() {
+  local a b
+  a=$1
+  for b in "$@"; do
+      if [ "${b}" -le "${a}" ]; then
+          a="${b}"
+      fi
+  done
+  echo "${a}"
+}
+SECONDS=0
+
+# General configuration
+stage=1              # Processes starts from the specified stage.
+stop_stage=10000     # Processes is stopped at the specified stage.
+skip_data_prep=false # Skip data preparation stages.
+skip_train=false     # Skip training stages.
+skip_eval=false      # Skip decoding and evaluation stages.
+skip_packing=true    # Skip the packing stage.
+skip_upload_hf=true  # Skip uploading to huggingface stage.
+ngpu=1               # The number of gpus ("0" uses cpu, otherwise use gpu).
+num_nodes=1          # The number of nodes.
+nodes=               # Node list if using ssh
+use_ssh=false        # Use SSH instead of cmd.sh
+nj=32                # The number of parallel jobs.
+dumpdir=dump         # Directory to dump features.
+expdir=exp           # Directory to save experiments.
+python=python3       # Specify python to execute espnet commands.
+
+# Data preparation related
+local_data_opts= # The options given to local/data.sh.
+
+# Speed perturbation related
+speed_perturb_factors=  # perturbation factors, e.g. "0.9 1.0 1.1" (separated by space).
+
+# Feature extraction related
+feats_type=raw       # Feature type (raw or fbank_pitch).
+audio_format=flac    # Audio format: wav, flac, wav.ark, flac.ark  (only in feats_type=raw).
+fs=16k               # Sampling rate.
+min_wav_duration=0.1 # Minimum duration in second.
+max_wav_duration=20  # Maximum duration in second.
+
+# Tokenization related
+token_type=word      # Tokenization type (char or bpe).
+token_list=
+# Pretrain model related
+ssl_args=         # Arguments for ssl model training, e.g., "--max_epoch 10".
+                     # Note that it will overwrite args in ssl config.
+num_splits_ssl=1 # Number of splitting for lm corpus.
+train_config=    # Configration files of training stage
+feats_normalize=
+
+# Upload model related
+hf_repo=
+inference_ssl_model=valid.loss.best.pth # SSL model path from previous iteration and uploading
+download_model= # Download a model from Model Zoo and use it for decoding.
+
+# [Task dependent] Set the datadir name created by local/data.sh
+train_set=     # Name of training set
+valid_set=     # Name of valid set
+lang=noinfo    # The language type of corpus.
+speech_fold_length=800 # fold_length for speech data during SSL training.
+text_fold_length=400   # fold_length for text data during SSL training.
+
+help_message=$(cat << EOF
+Usage: $0 --train-set "<train_set_name>" --valid-set "<valid_set_name>" --test_sets "<test_set_names>"
+
+Options:
```

Comment:
minor comment: please remember to align the help message to the above arguments.